services/ui-src/src/components/forms/SaveReturnButton.test.tsx

```diff
@@ -1,33 +1,68 @@
 import { render, screen } from "@testing-library/react";
 import userEvent from "@testing-library/user-event";
+import {
+  mockAdminUserStore,
+  mockStateUserStore,
+} from "utils/testing/mockZustand";
 import { SaveReturnButton } from "./SaveReturnButton";
+import { useStore } from "utils";
 
 const mockDisabledOnClick = jest.fn();
 const mockOnClick = jest.fn();
+const disabledOnClick = jest.fn();
+jest.mock("utils/state/useStore");
+const mockedUseStore = useStore as jest.MockedFunction<typeof useStore>;
 
 describe("<SaveReturnButton />", () => {
-  test("renders Save & return button", async () => {
-    render(<SaveReturnButton onClick={mockOnClick} />);
+  describe("<SaveReturnButton />", () => {
```

Comment:
nit: change this describe text to be more descriptive: is this block for state users?